internal/storage/image.go

```diff
@@ -897,6 +900,14 @@ func (st nativeStorageTransport) ResolveReference(ref types.ImageReference) (typ
 	return istorage.ResolveReference(ref)
 }
 
+// UpdatePinnedImagesList updates pinned images list in imageService.
+func (svc *imageService) UpdatePinnedImagesList(pauseImage string, pinnedImages []string) {
+	if pauseImage != "" {
+		pinnedImages = append(pinnedImages, pauseImage)
```

Comment:
If pause image is changed to ", what happens here? I forget, do we fallback to anything in sandbox_run_linux.go, and if so should we include it here?

Reply:
That's a good catch; I'll update it. :pray:

Reply:
I'm a bit confused; could you elaborate? 

AFAIU, the pause image in runPodSandbox comes from the config. Hence, it will be updated on HUP.  

https://github.com/cri-o/cri-o/blob/fbb899bb89a73917da0649a5c702964637454b32/server/sandbox_run_linux.go#L461-L464

Reply:
> If pause image is changed to "

This will not happen because the empty value in the config will not be parsed

> Unable to reload configuration: invalid reference format

Hence, `DefaultPauseImage` will always be in the list of `PinnedImages` and as `PauseImage`.

Reply:
Could you add an integration test for this?

Reply:
Just to make sure everything works as expected.

Reply:
I've implemented that checking here https://github.com/cri-o/cri-o/pull/7976/files#diff-4ea762d405a8752ac8b728bfbe94b8ee5f962013a1e65d4e5d20208556418447R255-R257